plugin/pkg/global/resourcereferencemanager/admission.go

```diff
@@ -581,6 +581,17 @@ func (r *ReferenceManager) ensureShootReferences(ctx context.Context, attributes
 		}
 	}
 
+	if shoot.Spec.DNS != nil {
```

Comment:
Can you only do the check if the shoot has no deletion timestamp? Otherwise, we might end up with stuck shoots as the `spec` is immutable for shoots in deletion.

Reply:
Nice catch.